### PR TITLE
feat : uniform not found error in data retrieval

### PR DESCRIPTION
**Pull request by zl03jsj**
- feat uniform `recode not found` error
- fix market-client retrieval-ask error on index not exists.

Fixes https://github.com/filecoin-project/venus/issues/4893

### CODE REVIEW

File: retrievalprovider/stream_handler.go [hunk 1]
```diff
@@ -69,13 +78,23 @@ func (p *RetrievalStreamHandler) HandleQueryStream(stream rmnet.RetrievalQuerySt
 		PieceCIDFound:   retrievalmarket.QueryItemUnavailable,
 		MinPricePerByte: big.Zero(),
 		UnsealPrice:     big.Zero(),
+		PaymentAddress:  p.paymentAddr,
 	}
 
 	minerDeals, err := p.pieceInfo.GetPieceInfoFromCid(ctx, query.PayloadCID, query.PieceCID)
 	if err != nil || len(minerDeals) == 0 {
```
Comment (diwufeiwen) [inline retrievalprovider/stream_handler.go:85]: 这里的建议if else拆分的更细节一点

File: retrievalprovider/piecestore.go [hunk 2]
```diff
@@ -42,5 +46,5 @@ func (pinfo *PieceInfo) GetPieceInfoFromCid(ctx context.Context, payloadCID cid.
 	if len(allMinerDeals) > 0 {
 		return allMinerDeals, nil
 	}
-	return nil, xerrors.Errorf("unable to find ready data for piece (%s) payload (%s)", piececid, payloadCID)
+	return nil, xerrors.Errorf("unable to find ready data for payload (%s), %w", piececid, payloadCID, repo.ErrNotFound)
```
Comment (diwufeiwen) [inline retrievalprovider/piecestore.go:49]: 占位符是不是不够

Comment (zl03jsj) [reply retrievalprovider/piecestore.go]: done

File: retrievalprovider/stream_handler.go (hunk 1 above)
Comment (zl03jsj) [reply retrievalprovider/stream_handler.go]: done

**Review by diwufeiwen — APPROVED** [review]
LGTM

**Pull request merged**